content/v2.0/application/settings.md

```diff
@@ -0,0 +1,266 @@
+---
+title: Settings
+order: 10
+---
+
+Hanami supports application settings through a `Settings` class, defined in `config/settings.rb`.
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    # Define your app settings here, for example:
+    #
+    # setting :my_flag, default: false, constructor: Types::Params::Bool
+  end
+end
+```
+
+Using this class, you can define what settings exist within your application, what types and defaults they have, and whether or not they are required for your application to boot.
+
+Each setting is read from an environment variable matching its name. For example, with the following configuration, the Redis URL and Sentry DSN are sourced from the `REDIS_URL` and `SENTRY_DSN` environment variables.
+
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :redis_url, constructor: Types::String
+    setting :sentry_dsn, constructor: Types::String
+  end
+end
+```
+
+## Types
+
+Environment variables are strings, but it's convenient for settings like `analytics_enabled` to be a boolean value, or `max_cart_items` to be an integer.
+
+You can coerce settings to these types by specifying the relevant constructor. For example, `Types::Params::Bool` and `Types::Params::Integer` will coerce values to boolean and integer values respectively:
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :analytics_enabled, constructor: Types::Params::Bool
+    setting :max_cart_items, constructor: Types::Params::Integer
+  end
+end
+```
+
+We can see this in action in the Hanami console:
+
+```shell
+ANALYTICS_ENABLED=true MAX_CART_ITEMS=100 bundle exec hanami console
+
+bookshelf[development]> Hanami.app["settings"].analytics_enabled
+=> true
+
+bookshelf[development]> Hanami.app["settings"].max_cart_items
+=> 100
+```
+
+Common types that are useful in settings constructors include:
+
+```ruby
+Types::String
+Types::Params::Bool
+Types::Params::Integer
+Types::Params::Float
+Types::Params::Date
+Types::Params::Time
+```
+
+## Required and optional settings
+
+Whether or not each setting is required for your application to start is determined by its constructor.
```

Comment:
```suggestion
Whether or not each setting is required for your application to boot is determined by its constructor.
```

Let's be consistent with "boot" instead of start?

Reply:
Adjusted.